optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/testdata/invalid/domain/TestdataInvalidQuarkusSolution.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.quarkus.testdata.invalid.domain;
+
+import java.util.List;
+
+import org.optaplanner.core.api.domain.solution.PlanningEntityCollectionProperty;
+import org.optaplanner.core.api.domain.solution.PlanningScore;
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.domain.solution.ProblemFactCollectionProperty;
+import org.optaplanner.core.api.domain.valuerange.ValueRangeProvider;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+
+@PlanningSolution
+public class TestdataInvalidQuarkusSolution {
```

Comment:
Naming wise, having these 3 testdata's prefixed with only "InvalidQuarkus" will clash in the long run with other fail fast checks.

I wouldn't mind a long name here to avoid that clashing. How about:

TestdataInvalidInverseRelationValue
TestdataInvalidInverseRelationEntity
TestdataInvalidInverseRelationSolution

Question: can we put these testdata classes in optaplanner-core,
add a "type=test-jar" and "scope=test" dependency from optaplanner-quarkus to optaplanner-core
and then have a quarkus test unit that adds these classes (think shrinkwrap)?

Reply:
Currently, we don't test invalid usage of annotations in core, so it will be confusing why there is a testdata domain in core that is not used in tests. It also increases the cognitive distance for the Quarkus test, since they need to look at core in order to see the domain. +1 for name change